src/packages/admin-ui-components/src/detail-panel/fields/boolean-field.tsx

```diff
@@ -11,6 +12,15 @@ export const BooleanField = ({
 	autoFocus: boolean;
 	disabled?: boolean;
 }) => {
+	useDataTransform({
```

Comment:
I removed this code in the previous PR as I was trying to flatten out the values stored in the form. It turns out that change is a lot more invasive than I thought, the combobox is really hard wired to store arrays of items. I've tested this across a range of scenarios and this is actually needed